src/web/mod.rs

```diff
@@ -270,6 +273,47 @@ pub(crate) fn build_axum_app(
     ))
 }
 
+pub(crate) fn build_axum_app(
+    context: &dyn Context,
+    template_data: Arc<TemplateData>,
+) -> Result<AxumRouter, Error> {
+    apply_middleware(routes::build_axum_routes(), context, Some(template_data))
+}
+
+pub(crate) fn build_metrics_axum_app(context: &dyn Context) -> Result<AxumRouter, Error> {
+    apply_middleware(routes::build_metric_routes(), context, None)
+}
+
+pub fn start_background_metrics_webserver(
+    addr: Option<&str>,
+    context: &dyn Context,
+) -> Result<(), Error> {
+    let axum_addr: SocketAddr = addr.unwrap_or(DEFAULT_BIND).parse()?;
+
+    tracing::info!(
+        "Starting metrics web server on `{}:{}`",
+        axum_addr.ip(),
+        axum_addr.port()
+    );
+
+    let metrics_axum_app = build_metrics_axum_app(context)?.into_make_service();
+    let runtime = context.runtime()?;
+
+    thread::spawn(move || {
```

Comment:
Do we need to spawn a thread, or can we just `runtime.spawn` this given that the runtime is kept alive by the context?

Reply:
valid point, updated